### PR TITLE
check usage on first request

### PR DESCRIPTION
**Pull request by ajmers**
We tally the calls we make to the Marketo API and check every 250 calls to make sure we haven't exceeded the rate limit. We need to check at the beginning of the job as well in case we previously exceeded the limit so that we don't make 250 calls before checking.

### CODE REVIEW

File: tap_marketo/__init__.py [hunk 1]
```diff
@@ -54,14 +55,16 @@ def get_start(entity):
...
+                      400 <= e.response.status_code < 500,
                       factor=2)
+
 def request(endpoint, params=None):
     if not CONFIG['token_expires'] or datetime.datetime.utcnow() >= CONFIG['token_expires']:
         refresh_token()
 
     CONFIG['call_count'] += 1
-    if CONFIG['call_count'] % 250 == 0:
+    if CONFIG['call_count'] % 250 == 0 or CONFIG['call_count'] == 1:
```
Comment (karstendick) [inline tap_marketo/__init__.py]: A simpler check would be:

```python
if CONFIG['call_count'] % 250 == 1:
```

File: tap_marketo/__init__.py [hunk 2]
```diff
@@ -299,28 +303,28 @@ def do_sync():
...
     for activity_type_id in LEADS_CHANGED_IDS:
         activity_type_ids.remove(activity_type_id)
-        logger.info("Syncing lead-altering activity type %d", activity_type_id)
+        LOGGER.info("Syncing lead-altering activity type %d", activity_type_id)
         sync_activities(activity_type_id, lead_fields, date_fields, leads_schema, do_leads=True)
 
     # If there are any unsynced leads after the last mutating activity type,
     # sync them before continuing
-    if len(LEAD_IDS) > 0:
+    if LEAD_IDS > 0:
```
Comment (karstendick) [inline tap_marketo/__init__.py]: This change does make sense. What does it mean to compare a set of things with the number 0?

File: tap_marketo/__init__.py (hunk 1 above)
Comment (ajmers) [inline tap_marketo/__init__.py]: good idea :man_facepalming:

File: tap_marketo/__init__.py (hunk 2 above)
Comment (ajmers) [inline tap_marketo/__init__.py]: good catch, this is meant to be `if LEAD_IDS:` to adhere to the `len-as-condition` pylint rule.

**Pull request merged**